src/com/madalinb/coffeeshop/Main.kt

```diff
@@ -0,0 +1,70 @@
+package com.madalinb.coffeeshop
+
+import com.madalinb.coffeeshop.csv.CSVParser
+import com.madalinb.coffeeshop.csv.CSVValidator
+import com.madalinb.coffeeshop.data.CSVFileOperations
+import java.math.RoundingMode
+import java.text.DecimalFormat
+
+fun main(args: Array<String>) {
+    val validator = CSVValidator()
+    val parser = CSVParser(validator)
+    val fileOps = CSVFileOperations(parser)
+
+    val map = mutableMapOf<Double, String>()
+    val df = DecimalFormat("#.####")
+    df.roundingMode = RoundingMode.CEILING
+
+    val x: Double? = args[0].toDoubleOrNull()
+    val y: Double? = args[1].toDoubleOrNull()
+    val filename = args[2]
+
+    if (args.size != 3) {
+        println("Error: incorrect number of application arguments")
+        showAppUsage()
+        return
+    }
+
+    if (x == null || y == null) {
+        println("Error: the coordinates for X and/or Y are incorrect")
+        showAppUsage()
+        return
+    }
+
+    if (filename.isEmpty()) {
+        println("Error: file name with the coffee shops data is missing")
+        showAppUsage()
+        return
+    }
+
+    val coffeeShops = fileOps.readFile(filename)
+
+    if (coffeeShops.isEmpty()) {
+        println("Processing the file '$filename' has found no valid entries")
+        return
+    }
+
+    var distance:Double
+    coffeeShops.forEach {
+        distance = it.distanceTo(x, y)
+        map[distance] = it.name
```

Comment:
This might not be correct, if you use a map and you have 2 shops at the same distance you will only show one to the user.

Reply:
Changed to List